Laura edits pmr 2 (#1888)
hashicorp/terraform-website: content/source/docs/cloud/registry/publish.html.md

  This allows the registry to inspect your module and generate documentation,
  track resource usage, and more.

- **`x.y.z` tags for releases:** At least one release tag must be present for you to publish a module. The registry uses release tags to identify module
  versions. Release tag names must be a [semantic version](http://semver.org),
  which can optionally be prefixed with a `v`. For example, `v1.0.4` and `0.9.2`. The registry ignores tags that do not look like version numbers.

## Publishing a New Module

You need [owners team](/docs/cloud/users-teams-organizations/permissions.html#organization-owners) permissions to publish private modules to the private module registry. You can publish modules through the UI as shown below or with the [Registry Modules API](../api/modules.html). The API also supports publishing modules without a VCS repo as the source, which is not possible via the UI.

To publish a new module:

1. Click "Registry" in the main navigation bar. Then click "Publish" and select "Module" in the dropdown.

    ![Terraform Cloud screenshot: the "registry" button and the "+Add Module" button](./images/publish-add-button.png)

>       The Add Module page appears with a list of available repositories.

Reviewer: ```suggestion
      The "Add Module" page appears with a list of available repositories.
```

      ![Terraform Cloud screenshot: the "add module" page, with a repository name entered](./images/publish-add-module.png)

2. Select the repository containing the module you want to publish.

>     You can search the list by typing all part or all of a repository name into the filter field. Remember that VCS providers use `<NAMESPACE>/<REPO NAME>` strings to locate repositories. The namespace is an organization name for most providers, but Bitbucket Server (not Bitbucket Cloud) uses project keys, like `INFRA`.

Reviewer: ```suggestion
    You can search the list by typing part or all of a repository name into the filter field. Remember that VCS providers use `<NAMESPACE>/<REPO NAME>` strings to locate repositories. The namespace is an organization name for most providers, but Bitbucket Server (not Bitbucket Cloud) uses project keys, like `INFRA`.
```

3. Click "Publish module".

    Terraform Cloud displays a loading page while it imports the module versions and then takes you to the new module's details page. On the details page, you can view available versions, read documentation, and copy a usage example.

      ![Terraform Cloud screenshot: a module details page](./images/publish-module-details.png)

## Releasing New Versions of a Module

To release a new version of a module, push a new release tag to its VCS repository. The registry will automatically import the new version.

Refer to [Preparing a Module Repository](#preparing-a-module-repository) for details about release tag requirements.

## Deleting Versions and Modules

> You can delete individual versions of a module or entire modules. If a deletion would leave a module with no versions, Terraform Cloud will remove the entire module. To delete a module or version:

Reviewer: ```suggestion
You can delete individual versions of a module or the entire module. If deleting a module version would leave a module with no versions, Terraform Cloud will remove the entire module. To delete a module or version:
```

1. Navigate to the module's details page.
2. If you want to delete a single version, use the "Versions" dropdown to select it.
3. Click "Delete module".
4. Select the desired action from the dropdown:
   - **Delete only this module version:** Deletes only the version of the module you were viewing when you clicked "Delete module".
   - **Delete all versions for this provider for this module:** Deletes the entire module for a single provider. This is important if you have modules with the same name but with different providers. For example, if you have module repos named `terraform-aws-appserver` and `terraform-azure-appserver`, the registry treats them as alternate providers of the same `appserver` module.
   - **Delete all providers and versions for this module:** Deletes all modules with this name, even if they are from different providers. For example, both `terraform-aws-appserver` and `terraform-azure-appserver` would be deleted.

    ![Terraform Cloud screenshot: the deletion dialog](./images/publish-delete.png)

5. Type the module name and click "Delete".

> 
> 
> 
> ### Restoring a Deleted Module or Version

Reviewer: ```suggestion
### Restoring a Deleted Module or Version

```
